src/components/Router/index.js

```diff
@@ -42,10 +42,18 @@ class CrossDeviceMobileRouter extends Component {
 
   setConfig = (actions) => (data) => {
     const {token, steps, documentType, step} = data
-    this.setState({token, steps, step})
+    const crossDeviceClientSteps = this.replaceLastStep(steps)
+    this.setState({token, steps: crossDeviceClientSteps, step})
     actions.setDocumentType(documentType)
   }
 
+  replaceLastStep = (steps) => {
+    const filteredSteps = steps.filter(step => {
```

Comment:
I don't think you need to go through every step - it's only the last one you are interested in.

I would also move this logic to `stepComponentMap` - you could just pass the `mobileFlow` option in

Reply:
True, I will do an array pop if 'complete' step is present.

I think this logic belongs here because everywhere else we pass the right steps from the parent router (in this case `CrossDeviceMobileRouter`) to the HistoryRouter. If I put this logic in the stepComponentMap I will have to add a component to update the state.steps both in the parent router and in the history router as there might be some functions using the state steps as the source of truth.